docker/docker.go

```diff
@@ -84,12 +84,6 @@ func main() {
 
 	setDefaultConfFlag(flTrustKey, defaultTrustKeyFile)
 
-	// Regardless of whether the user sets it to true or false, if they
-	// specify --tlsverify at all then we need to turn on tls
-	if flag.IsSet("-tlsverify") {
```

Comment:
@cpuguy83 i believe we need to keep this here. The fix could be though: `if flag.IsSet("-tlsverify") || *flTlsVerify {`. Looking in history to understand better what happened.